src/environment.rs

```diff
@@ -0,0 +1,59 @@
+#![allow(dead_code)]
+
+use std::collections::HashMap;
+
+use anyhow::bail;
+
+use crate::{matrices::Matrix, traits::MatrixNumber};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Identifier {
+    id: String,
+}
+
+impl Identifier {
+    pub fn new(id: &str) -> anyhow::Result<Self> {
```

Comment:
lepiej żeby brało String, a nie referencję, bo i tak w środku ta funkcja dokonuje alokacji w obecnej wersji, więc lepiej żeby było to explicit